### PR TITLE
chore: remove components not part of upcoming releases

### PR DESCRIPTION
**Pull request by dierat**
### Summary:
I'm trying to clear out the cruft in the `next` branch to get it ready so it can cleanly merge into `main` when the CAD project is finished. In this PR, I'm removing components that are not:
- currently published on `main`
- being used internally (ie in storybook for presentation purposes)
- being used in the CAD project
- part of the Project Overview project that was done before CAD

My plan is to move the Project Overview work onto another branch off of `next` so that work can be referenced by teams but it's not actually shipped with the design system. If we do want that stuff on `main`, it can be merged separately ahead of CAD.

For which components are being released as part of CAD, I've been going off of this list: https://app.shortcut.com/czi-edu/story/196002/clarify-the-list-of-components-included-in-the-first-release (Keep in mind that list does not consider dependencies; there will probably be other subcomponents etc that will be part of the project but not in the main list.)

### Test Plan:
Storybook loads fine,
no type errors,
no chromatic changes (aside from components being removed),
and none of the components removed in this PR are in the plan for CAD components: 

### List of component deleted in this PR:
The ~54~ 53 component being deleted here are:
- Accordion
- AccordionPanel
- Band
- CheckboxField
- CheckboxFieldItem
- Counter
- DefinitionList
- DefinitionListItem
- Drawer
- DrawerBody
- DrawerFooter
- DrawerHeader
- Feature
- FieldNote
- FileUploadField
- Footer
- FooterNav
- FooterNavItem
- Hero
- InlineForm
- KeyValueTable
- Label
- LoadingIndicator
- MediaBlock
- Modal
- ModalBody
- ModalFooter
- ModalHeader
- Radio
- RadioField
- RadioFieldItem
- ~ShowHide~
- SkeletonBar
- StackedBlock
- Table
- TableBody
- TableCell
- TableFooter
- TableHeader
- TableHeaderCell
- TableObject
- TableObjectBody
- TableObjectFooter
- TableObjectHeader
- TableRow
- Tags
- Textarea
- TextareaField
- TextField
- TextInput
- TextList
- Tile
- Toolbar
- ToolbarItem

### CODE REVIEW

File: src/components/ShowHide/ShowHide.tsx [hunk 1]
```diff
@@ -1,127 +0,0 @@
-/* eslint-disable jsx-a11y/no-static-element-interactions */
```
Comment (dierat) [inline src/components/ShowHide/ShowHide.tsx:1]: Looks like `ShowHide` might be part of Project Overview? https://app.shortcut.com/czi-edu/story/190248/showhide cc @ifrost1

**Comment by dierat** [comment]
Another idea from @anniehu4 is to, instead of deleting these components, we could move them to another folder, remove them from storybook, and remove their exports in `src/index.ts` so they're not accessible outside of the system.

@anniehu4 were you thinking something like..

- `src`
  - `components` (stuff people can use outside of EDS)
  - `upcomingComponents` (these components)
  - `index.ts` (only lists components from the `components` directory)

?

**Review by anniehu4 — APPROVED** [review]
> were you thinking something like..
src
components (stuff people can use outside of EDS)
upcomingComponents (these components)
index.ts (only lists components from the components directory)
https://github.com/chanzuckerberg/edu-design-system/compare/next...diedra/remove-later-release-components-2

yess exactly! I think if we update `.storybook/main.js` to only look in `src/components` instead of all of `src/` that'll take care of hiding the stories. super nit is we seem to be kebab-casing folder names, so it'd be `upcoming-components/`

**Comment by dierat** [comment]
> > were you thinking something like..
> > 
> > * src
> >   - components (stuff people can use outside of EDS)
> >   - upcomingComponents (these components)
> >   - index.ts (only lists components from the components directory)
> 
> > [next...diedra/remove-later-release-components-2](https://github.com/chanzuckerberg/edu-design-system/compare/next...diedra/remove-later-release-components-2)
> 
> yess exactly! I think if we update `.storybook/main.js` to only look in `src/components` instead of all of `src/` that'll take care of hiding the stories. super nit is we seem to be kebab-casing folder names, so it'd be `upcoming-components/`

Ah, right, good calls. How does this look? https://github.com/chanzuckerberg/edu-design-system/pull/1047

**Comment by anniehu4** [comment]
it looks perfect ❤️  we'll close this one, yeah?

**Comment by dierat** [comment]
Work is continuing in https://github.com/chanzuckerberg/edu-design-system/pull/1047

**Pull request closed without merging**